src/client/common/client_common.cpp

```diff
@@ -76,28 +109,55 @@ std::string mp::cmd::update_notice(const mp::UpdateInfo& update_info)
                                      update_info.description(), update_info.url()));
 }
 
-std::shared_ptr<grpc::Channel> mp::client::make_channel(const std::string& server_address,
-                                                        mp::RpcConnectionType conn_type,
-                                                        mp::CertProvider& cert_provider)
+std::shared_ptr<grpc::Channel> mp::client::make_secure_channel(const std::string& server_address,
+                                                               mp::CertProvider* cert_provider)
 {
-    std::shared_ptr<grpc::ChannelCredentials> creds;
-    if (conn_type == mp::RpcConnectionType::ssl)
-    {
-        auto opts = grpc::SslCredentialsOptions();
-        opts.server_certificate_request = GRPC_SSL_REQUEST_SERVER_CERTIFICATE_BUT_DONT_VERIFY;
-        opts.pem_cert_chain = cert_provider.PEM_certificate();
-        opts.pem_private_key = cert_provider.PEM_signing_key();
-        creds = grpc::SslCredentials(opts);
-    }
-    else if (conn_type == mp::RpcConnectionType::insecure)
+    if (!cert_provider)
     {
-        creds = grpc::InsecureChannelCredentials();
-    }
-    else
-    {
-        throw std::runtime_error("Unknown connection type");
+        auto data_location{MP_STDPATHS.writableLocation(StandardPaths::GenericDataLocation)};
+        auto common_client_cert_dir_path{data_location + common_client_cert_dir};
+
+        // The following logic is for determing which certificate to use when the client starts up.
+        // 1. First check if the new common client certificate exists and use it if it does.
+        // 2. Failing that, check if the multipass-gui certificate exists and determine if it's authenticated
+        //    with the daemon already.  If it is, copy it to the common client certificate directory and use it.
+        // 3. If that fails, then try the certificate from the cli client in the same manner as 2.
+        // 4. Lastly, no known certificate for the user exists, so create a new common certificate and use that.
```

Comment:
And remove the old certs?

Reply:
I suppose we could :shrug: Just adds more throw away code to save a few bytes on the user's system.

Reply:
IMO that'd be prudent to do, yeah.

Reply:
Ok, can do.

Reply:
This is implemented now.